src/meta/src/controller/streaming_job.rs

```diff
@@ -541,12 +541,32 @@ impl CatalogController {
             return Err(MetaError::permission_denied("table version is stale"));
         }
 
+        // 2. check concurrent replace.
+        let replacing_cnt = ObjectDependency::find()
+            .join(
+                JoinType::InnerJoin,
+                object_dependency::Relation::Object1.def(),
+            )
+            .join(JoinType::InnerJoin, object::Relation::StreamingJob.def())
+            .filter(
+                object_dependency::Column::Oid
+                    .eq(id as ObjectId)
+                    .and(object::Column::ObjType.eq(ObjectType::Table))
+                    .and(streaming_job::Column::JobStatus.ne(JobStatus::Created)),
+            )
+            .count(&txn)
+            .await?;
+        if replacing_cnt != 0 {
+            assert_eq!(replacing_cnt, 1);
+            return Err(MetaError::permission_denied("table is being altered"));
+        }
```

Comment:
Is it possible we have a `create mv on table` blocking the `alter` since you are checking the existing table object in streaming job

Reply:
Oh yes, 👍  after some tests I found that `alter` will make `create mv on table` fail and trigger recovery if it's under barrier collection step. So I think `create mv on table` should be mutually exclusive with `block` alter as well. Let me remove the assertion and add the check in create side as well.

Reply:
Updated. The following query will be used for `create mv on table`:
```sql
SELECT COUNT(*) AS num_items
FROM
  (SELECT "object_dependency"."id",
          "object_dependency"."oid",
          "object_dependency"."used_by"
   FROM "object_dependency"
   INNER JOIN "object" ON "object_dependency"."used_by" = "object"."oid"
   INNER JOIN "streaming_job" ON "object"."oid" = "streaming_job"."job_id"
   WHERE "object_dependency"."oid" IN (6)
     AND "object"."obj_type" = 'TABLE'
     AND "streaming_job"."job_status" <> 'CREATED'
     AND "object"."oid" NOT IN
       (SELECT "table_id"
        FROM "table") ) AS "sub_query"
```